src/lib/services/api-token-service.ts

```diff
@@ -160,7 +160,9 @@ export class ApiTokenService {
         }
     }
 
-    public getUserForToken(secret: string): IApiUser | undefined {
+    public async getUserForToken(
+        secret: string,
+    ): Promise<IApiUser | undefined> {
```

Comment:
We're marking this async, doesn't this want resolves on its non promise return values now?

Reply:
Not sure I understand your question. All async methods needs to return a Promise as far as I know, which is why now we have to do awaits every time we call this method. E.g.: https://github.com/Unleash/unleash/blob/654a880b4f1584432bc3a9a553c711d67d87ef85/src/lib/routes/auth/reset-password-controller.ts#L167

Reply:
You're absolutely right, I forget that JS has this syntactic cancer